web/ui/react-app/src/index.tsx

```diff
@@ -4,4 +4,16 @@ import ReactDOM from 'react-dom';
 import App from './App';
 import 'bootstrap/dist/css/bootstrap.min.css';
 
-ReactDOM.render(<App />, document.getElementById('root'));
+// Declared/defined in public/index.html, value replaced by Prometheus when serving bundle.
+declare const PATH_PREFIX: string;
+
+let prefix = PATH_PREFIX;
+if (PATH_PREFIX === 'PATH_PREFIX_PLACEHOLDER' || PATH_PREFIX === '/') {
+  // Either we are running the app outside of Prometheus, so the placeholder value in
+  // the index.html didn't get replaced, or we have a '/' prefix, which we also need to
+  // normalize to '' to make concatenations work (prefixes like '/foo/bar/' already get
+  // their trailing slash stripped by Prometheus).
+  prefix = '';
+}
+
```

Comment:
Just an idea - instead of passing prefix down to all components can you just export from here all the endpoints
For example, Status page endpoints will look like:
```
export const statusPagePaths = [
    '/api/v1/status/runtimeinfo',
    '/api/v1/status/buildinfo',
    '/api/v1/alertmanagers'
].map(p => `${prefix}${p}`);

```

I think this will reduce a lot the number of changes. @juliusv  WDYT?

Reply:
Yeah, but I would very much like to avoid global state whereever possible (that's why I only use the global path prefix here in the `index.tsx` and then never again). Passing in dependencies into components is more verbose, but in exchange we get unambiguous explicitness, with dependencies enumerated as properties to components.

Reply:
Another approach can be to pass the prefix(explicitly) as Link state prop. Then prefix is available through props in each component ruled by Reach router

Reply:
Chatted with @boyskila to find out what was meant here. https://reach.tech/router/api/Link provides a way to attach location state to the Reach router, but it doesn't seem to be quite for this kind of use case.

Reply:
I also would like to avoid global state whenever possible, and like to see dependencies clearly defined in the props.